Replacing: slf4j with jul

Ant-hem opened the pull request:
Replaced SLF4J with Java.Util.logging for tracing requests/responses

Removing slf4j allow to remove on more dependency of the library and
to simplify the shading process.

 - Added: JUL in HttpTransport with Log.FINEST level
 - Removed: SLF4J and Logback from pom.xml
 - Updated: logging.properties

 To activate JUL please set in logging.properties
 .level=FINEST
 com.algolia.search.HttpTransport=FINEST
 java.util.logging.ConsoleHandler.level=FINEST

 and set:
 -Djava.util.logging.config.file="logging.properties-path"

Ant-hem committed:
Replaced SLF4J with Java.Util.logging for tracing requests/responses

Removing slf4j allow to remove on more dependency of the library and
to simplify the shading process.

 - Added: JUL in HttpTransport with Log.FINEST level
 - Removed: SLF4J and Logback from pom.xml
 - Updated: logging.properties

 To activate JUL please set in logging.properties
 .level=FINEST
 com.algolia.search.HttpTransport=FINEST
 java.util.logging.ConsoleHandler.level=FINEST

 and set:
 -Djava.util.logging.config.file="logging.properties-path"

 More information about JUL:
 http://java.ociweb.com/mark/programming/JavaLogging.html

BenoitPerrot commented on algoliasearch-core/src/main/java/com/algolia/search/HttpTransport.java:
```diff
@@ -27,7 +27,7 @@
   private final RetryStrategy retryStrategy;
   private final ConfigBase config;
-  private static final Logger logger = LoggerFactory.getLogger("algoliasearch");
+  private static final Logger LOGGER = Logger.getLogger(HttpTransport.class.getName());
```
For another commit: ensure all static-final are uppercase:
- make javaVersion uppercase in Config.base
- if possible add a rule in the formatting configuration

BenoitPerrot commented on algoliasearch-core/src/test/resources/logging.properties:
```diff
@@ -1,8 +1,26 @@
+com.algolia.search.HttpTransport=INFO
 org.junit.platform.launcher.core.LauncherConfigurationParameters.level=WARNING
 org.junit.jupiter.engine.config.EnumConfigurationParameterConverter.level=WARNING
+org.apache.http.level=WARNING
```
-1: Add back the carriage return at the end of the line

BenoitPerrot commented on pom.xml:
```diff
@@ -208,11 +201,6 @@
                             <configuration>
                                 <shadedClassifierName>uber</shadedClassifierName>
                                 <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
```
-1: Remove this promoteTransitiveDependencies, as now unnecessary

This was introduced to have "algoliasearch-apache-uber" list "slf4j-api" as a dependency in its reduced pom.

Ant-hem committed:
Fixed: static naming conventions in configbase

Ant-hem commented:
All good @BenoitPerrot ! I will submit another PR with the formatting rules later :)

BenoitPerrot commented on algoliasearch-core/src/test/resources/logging.properties:
```diff
@@ -1,8 +1,26 @@
+com.algolia.search.HttpTransport=INFO
 org.junit.platform.launcher.core.LauncherConfigurationParameters.level=WARNING
 org.junit.jupiter.engine.config.EnumConfigurationParameterConverter.level=WARNING
+org.apache.http.level=WARNING
```
Hmm 🤔 This shouldn't be necessary

Ant-hem replied:
Good catch 😄

Ant-hem committed:
Added: logging.properties in algoliasearch-apache

- Removed org.apache.http.level=WARNING from algoliasearch-core
- Added org.apache.http.level=WARNING in algoliasearch-apache file
